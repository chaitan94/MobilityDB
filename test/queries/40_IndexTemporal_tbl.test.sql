-------------------------------------------------------------------------------

VACUUM ANALYZE tbl_tbool_big;
VACUUM ANALYZE tbl_tint_big;
VACUUM ANALYZE tbl_tfloat_big;
VACUUM ANALYZE tbl_ttext_big;

DROP INDEX IF EXISTS tbl_tbool_big_gist_idx;
DROP INDEX IF EXISTS tbl_tbool_big_spgist_idx;

DROP INDEX IF EXISTS tbl_tint_big_gist_idx;
DROP INDEX IF EXISTS tbl_tint_big_spgist_idx;

DROP INDEX IF EXISTS tbl_tfloat_big_gist_idx;
DROP INDEX IF EXISTS tbl_tfloat_big_spgist_idx;

DROP INDEX IF EXISTS tbl_ttext_big_gist_idx;
DROP INDEX IF EXISTS tbl_ttext_big_spgist_idx;

-------------------------------------------------------------------------------

CREATE INDEX tbl_tbool_big_gist_idx ON tbl_tbool_big USING GIST(temp);
CREATE INDEX tbl_tint_big_gist_idx ON tbl_tint_big USING GIST(temp);
CREATE INDEX tbl_tfloat_big_gist_idx ON tbl_tfloat_big USING GIST(temp);
CREATE INDEX tbl_ttext_big_gist_idx ON tbl_ttext_big USING GIST(temp);

SELECT count(*) FROM tbl_tbool_big WHERE temp && NULL::period;
SELECT count(*) FROM tbl_tbool_big WHERE temp && NULL::tbool;
SELECT count(*) FROM tbl_tbool_big WHERE temp && tbool '[true@2001-01-01, true@2001-07-01]';

SELECT count(*) FROM tbl_tbool_big WHERE temp && period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tbool_big WHERE temp @> period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tbool_big WHERE temp <@ period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tbool_big WHERE temp ~= period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tbool_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tbool_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tbool_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tbool_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';

SELECT count(*) FROM tbl_tint_big WHERE temp && intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp @> intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp <@ intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp ~= intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp << intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp &< intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp >> intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp &> intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tint_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tint_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tint_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';

SELECT count(*) FROM tbl_tfloat_big WHERE temp && floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp @> floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp <@ floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp ~= floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp << floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp &< floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp >> floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp &> floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';

SELECT count(*) FROM tbl_ttext_big WHERE temp && period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_ttext_big WHERE temp @> period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_ttext_big WHERE temp <@ period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_ttext_big WHERE temp ~= period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_ttext_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_ttext_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_ttext_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_ttext_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';

DROP INDEX IF EXISTS tbl_tbool_big_gist_idx;
DROP INDEX IF EXISTS tbl_tint_big_gist_idx;
DROP INDEX IF EXISTS tbl_tfloat_big_gist_idx;
DROP INDEX IF EXISTS tbl_ttext_big_gist_idx;

-------------------------------------------------------------------------------

CREATE INDEX tbl_tbool_big_spgist_idx ON tbl_tbool_big USING SPGIST(temp);
CREATE INDEX tbl_tint_big_spgist_idx ON tbl_tint_big USING SPGIST(temp);
CREATE INDEX tbl_tfloat_big_spgist_idx ON tbl_tfloat_big USING SPGIST(temp);
CREATE INDEX tbl_ttext_big_spgist_idx ON tbl_ttext_big USING SPGIST(temp);

SELECT count(*) FROM tbl_tbool_big WHERE temp && NULL::period;
SELECT count(*) FROM tbl_tbool_big WHERE temp && NULL::tbool;
SELECT count(*) FROM tbl_tbool_big WHERE temp && tbool '[true@2001-01-01, true@2001-07-01]';

SELECT count(*) FROM tbl_tbool_big WHERE temp && period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tbool_big WHERE temp @> period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tbool_big WHERE temp <@ period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tbool_big WHERE temp ~= period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tbool_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tbool_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tbool_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tbool_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';

SELECT count(*) FROM tbl_tint_big WHERE temp && intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp @> intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp <@ intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp ~= intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp << intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp &< intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp >> intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp &> intrange '[1,50]';
SELECT count(*) FROM tbl_tint_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tint_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tint_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tint_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';

SELECT count(*) FROM tbl_tfloat_big WHERE temp && floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp @> floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp <@ floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp ~= floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp << floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp &< floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp >> floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp &> floatrange '[1,50]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_tfloat_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';

SELECT count(*) FROM tbl_ttext_big WHERE temp && period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_ttext_big WHERE temp @> period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_ttext_big WHERE temp <@ period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_ttext_big WHERE temp ~= period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_ttext_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_ttext_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_ttext_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
SELECT count(*) FROM tbl_ttext_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';

DROP INDEX IF EXISTS tbl_tbool_big_spgist_idx;
DROP INDEX IF EXISTS tbl_tint_big_spgist_idx;
DROP INDEX IF EXISTS tbl_tfloat_big_spgist_idx;
DROP INDEX IF EXISTS tbl_ttext_big_spgist_idx;


-------------------------------------------------------------------------------

