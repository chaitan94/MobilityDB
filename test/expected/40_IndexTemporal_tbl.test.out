VACUUM ANALYZE tbl_tbool_big;
VACUUM
VACUUM ANALYZE tbl_tint_big;
VACUUM
VACUUM ANALYZE tbl_tfloat_big;
VACUUM
VACUUM ANALYZE tbl_ttext_big;
VACUUM
DROP INDEX IF EXISTS tbl_tbool_big_gist_idx;
NOTICE:  index "tbl_tbool_big_gist_idx" does not exist, skipping
DROP INDEX
DROP INDEX IF EXISTS tbl_tbool_big_spgist_idx;
NOTICE:  index "tbl_tbool_big_spgist_idx" does not exist, skipping
DROP INDEX
DROP INDEX IF EXISTS tbl_tint_big_gist_idx;
NOTICE:  index "tbl_tint_big_gist_idx" does not exist, skipping
DROP INDEX
DROP INDEX IF EXISTS tbl_tint_big_spgist_idx;
NOTICE:  index "tbl_tint_big_spgist_idx" does not exist, skipping
DROP INDEX
DROP INDEX IF EXISTS tbl_tfloat_big_gist_idx;
NOTICE:  index "tbl_tfloat_big_gist_idx" does not exist, skipping
DROP INDEX
DROP INDEX IF EXISTS tbl_tfloat_big_spgist_idx;
NOTICE:  index "tbl_tfloat_big_spgist_idx" does not exist, skipping
DROP INDEX
DROP INDEX IF EXISTS tbl_ttext_big_gist_idx;
NOTICE:  index "tbl_ttext_big_gist_idx" does not exist, skipping
DROP INDEX
DROP INDEX IF EXISTS tbl_ttext_big_spgist_idx;
NOTICE:  index "tbl_ttext_big_spgist_idx" does not exist, skipping
DROP INDEX
CREATE INDEX tbl_tbool_big_gist_idx ON tbl_tbool_big USING GIST(temp);
CREATE INDEX
CREATE INDEX tbl_tint_big_gist_idx ON tbl_tint_big USING GIST(temp);
CREATE INDEX
CREATE INDEX tbl_tfloat_big_gist_idx ON tbl_tfloat_big USING GIST(temp);
CREATE INDEX
CREATE INDEX tbl_ttext_big_gist_idx ON tbl_ttext_big USING GIST(temp);
CREATE INDEX
SELECT count(*) FROM tbl_tbool_big WHERE temp && NULL::period;
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp && NULL::tbool;
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp && tbool '[true@2001-01-01, true@2001-07-01]';
 count 
-------
  4840
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp && period '[2001-01-01, 2001-07-01]';
 count 
-------
  4840
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp @> period '[2001-01-01, 2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp <@ period '[2001-01-01, 2001-07-01]';
 count 
-------
  4839
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp ~= period '[2001-01-01, 2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
 count 
-------
     1
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
 count 
-------
  4840
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
 count 
-------
  4759
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';
 count 
-------
  9599
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp && intrange '[1,50]';
 count 
-------
  7857
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp @> intrange '[1,50]';
 count 
-------
   666
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp <@ intrange '[1,50]';
 count 
-------
  1924
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp ~= intrange '[1,50]';
 count 
-------
     2
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp << intrange '[1,50]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp &< intrange '[1,50]';
 count 
-------
  1924
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp >> intrange '[1,50]';
 count 
-------
  1743
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp &> intrange '[1,50]';
 count 
-------
  9600
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
 count 
-------
  4703
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
 count 
-------
  4897
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';
 count 
-------
  9600
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp && tint '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
  1952
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp @> tint '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp <@ tint '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
   162
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp ~= tint '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp && floatrange '[1,50]';
 count 
-------
  7825
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp @> floatrange '[1,50]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp <@ floatrange '[1,50]';
 count 
-------
  1728
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp ~= floatrange '[1,50]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp << floatrange '[1,50]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp &< floatrange '[1,50]';
 count 
-------
  1728
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp >> floatrange '[1,50]';
 count 
-------
  1775
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp &> floatrange '[1,50]';
 count 
-------
  9600
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
 count 
-------
     1
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
 count 
-------
  4753
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
 count 
-------
  4847
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';
 count 
-------
  9599
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp && tfloat '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
  1930
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp @> tfloat '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp <@ tfloat '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
   128
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp ~= tfloat '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp && period '[2001-01-01, 2001-07-01]';
 count 
-------
  4788
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp @> period '[2001-01-01, 2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp <@ period '[2001-01-01, 2001-07-01]';
 count 
-------
  4785
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp ~= period '[2001-01-01, 2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
 count 
-------
  4786
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
 count 
-------
  4812
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';
 count 
-------
  9599
(1 row)

DROP INDEX IF EXISTS tbl_tbool_big_gist_idx;
DROP INDEX
DROP INDEX IF EXISTS tbl_tint_big_gist_idx;
DROP INDEX
DROP INDEX IF EXISTS tbl_tfloat_big_gist_idx;
DROP INDEX
DROP INDEX IF EXISTS tbl_ttext_big_gist_idx;
DROP INDEX
CREATE INDEX tbl_tbool_big_spgist_idx ON tbl_tbool_big USING SPGIST(temp);
CREATE INDEX
CREATE INDEX tbl_tint_big_spgist_idx ON tbl_tint_big USING SPGIST(temp);
CREATE INDEX
CREATE INDEX tbl_tfloat_big_spgist_idx ON tbl_tfloat_big USING SPGIST(temp);
CREATE INDEX
CREATE INDEX tbl_ttext_big_spgist_idx ON tbl_ttext_big USING SPGIST(temp);
CREATE INDEX
SELECT count(*) FROM tbl_tbool_big WHERE temp && NULL::period;
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp && NULL::tbool;
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp && tbool '[true@2001-01-01, true@2001-07-01]';
 count 
-------
  4840
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp && period '[2001-01-01, 2001-07-01]';
 count 
-------
  4840
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp @> period '[2001-01-01, 2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp <@ period '[2001-01-01, 2001-07-01]';
 count 
-------
  4839
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp ~= period '[2001-01-01, 2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
 count 
-------
     1
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
 count 
-------
  4840
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
 count 
-------
  4759
(1 row)

SELECT count(*) FROM tbl_tbool_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';
 count 
-------
  9599
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp && intrange '[1,50]';
 count 
-------
  7857
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp @> intrange '[1,50]';
 count 
-------
   666
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp <@ intrange '[1,50]';
 count 
-------
  1924
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp ~= intrange '[1,50]';
 count 
-------
     2
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp << intrange '[1,50]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp &< intrange '[1,50]';
 count 
-------
  1924
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp >> intrange '[1,50]';
 count 
-------
  1743
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp &> intrange '[1,50]';
 count 
-------
  9600
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
 count 
-------
  4703
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
 count 
-------
  4897
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';
 count 
-------
  9600
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp && tint '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
  1952
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp @> tint '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp <@ tint '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
   162
(1 row)

SELECT count(*) FROM tbl_tint_big WHERE temp ~= tint '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp && floatrange '[1,50]';
 count 
-------
  7825
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp @> floatrange '[1,50]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp <@ floatrange '[1,50]';
 count 
-------
  1728
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp ~= floatrange '[1,50]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp << floatrange '[1,50]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp &< floatrange '[1,50]';
 count 
-------
  1728
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp >> floatrange '[1,50]';
 count 
-------
  1775
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp &> floatrange '[1,50]';
 count 
-------
  9600
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
 count 
-------
     1
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
 count 
-------
  4753
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
 count 
-------
  4847
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';
 count 
-------
  9599
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp && tfloat '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
  1930
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp @> tfloat '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp <@ tfloat '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
   128
(1 row)

SELECT count(*) FROM tbl_tfloat_big WHERE temp ~= tfloat '[1@2001-01-01, 10@2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp && period '[2001-01-01, 2001-07-01]';
 count 
-------
  4788
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp @> period '[2001-01-01, 2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp <@ period '[2001-01-01, 2001-07-01]';
 count 
-------
  4785
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp ~= period '[2001-01-01, 2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp <<# period '[2001-01-01, 2001-07-01]';
 count 
-------
     0
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp &<# period '[2001-01-01, 2001-07-01]';
 count 
-------
  4786
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp #>> period '[2001-01-01, 2001-07-01]';
 count 
-------
  4812
(1 row)

SELECT count(*) FROM tbl_ttext_big WHERE temp #&> period '[2001-01-01, 2001-07-01]';
 count 
-------
  9599
(1 row)

DROP INDEX IF EXISTS tbl_tbool_big_spgist_idx;
DROP INDEX
DROP INDEX IF EXISTS tbl_tint_big_spgist_idx;
DROP INDEX
DROP INDEX IF EXISTS tbl_tfloat_big_spgist_idx;
DROP INDEX
DROP INDEX IF EXISTS tbl_ttext_big_spgist_idx;
DROP INDEX
